browser/lib/util/http.js

```diff
@@ -3,6 +3,16 @@ var Http = (function() {
 
 	function Http() {}
 
+	function shouldFallback(err) {
+		var statusCode = err.statusCode;
+		/* 400 + no code = a generic xhr onerror. Browser doesn't give us enough
+		 * detail to know whether it's fallback-fixable, but it may be (eg if a
+		 * network issue), so try just in case */
+		return (statusCode === 408 && !err.code) ||
+			(statusCode === 400 && !err.code)      ||
```

Comment:
Tabs are for leading indentation only; never anywhere except at the start of a line. Then, spaces only where whitespace is needed for syntax/readability.

All other spaces are vanity spacing only :) If you do want to do that, it only ever applies among lines that have equal numbers of initial tabs. If things don't line up the way you like, deal with it :)